### ISSUE TITLE
Controller for custom logics?

### ISSUE THREAD

**Issue by lzzy12** [Mantano/iridium#78]
Hi, my use-case requires to get the page number the use currently is on and the total number of pages. Is this possible to get with the EpubReader.fromFile constructor or is there any other way to do this? I am looking for something like a controller class for the reade.

**Comment by jmgeffroy** [comment]
Hi @lzzy12, In fact, the ReaderContext does provide a `currentLocationStream`, which seems to be what you need if I understand correctly. To see an example in action, you can have a look at [this code in the ReaderToolbar class](https://github.com/jmgeffroy/iridium/blob/b7cde3b60154bdf47a085dc98b06e026a7306f57/reader_widget/lib/views/viewers/ui/reader_toolbar.dart#L48-L48).
If access to this stream is what you ned, I'll think about a clean way to provide access to it.

**Comment by lzzy12** [comment]
Yes it will be helpful if this could be available through an API outside the package.

**Comment by jmgeffroy** [comment]
I have added a `paginationCallback` parameter to the EpubScreen API, and adapted the reader widget example to showcase it:

```
        EpubScreen.fromPath(
          filePath: widget.dirPath,
          location: '{"cfi":" ","idref":"file_12.html"}',
          paginationCallback: (paginationInfo) {
            Fimber.d("--- paginationInfo: $paginationInfo");
          },
        )
```
The [PaginationInfo](https://github.com/Mantano/iridium/blob/main/components/navigator/lib/src/epub/callbacks/model/pagination_info.dart) class contains a lot of information about the position.

Does it match your requirements ?

**Comment by lzzy12** [comment]
Hi, thanks for your quick responses. I could not figure out which one of the attributes stores the total number of pages in the epub file. Could you point me to it?

**Comment by jmgeffroy** [comment]
Hello @lzzy12, in fact `PaginationInfo` contains a `page` method, which is defined as follows ([pagination_info.dart](https://github.com/jmgeffroy/iridium/blob/794cde45014ee7bd8f07d068ce399e2bdf8672bf/components/navigator/lib/src/epub/callbacks/model/pagination_info.dart#L46-L46)):

```
  int get page =>
      linkPagination.firstPageNumber +
      percent * (linkPagination.pagesCount - 1) ~/ 100;
```

**Comment by lzzy12** [comment]
Hi @jmgeffroy Currently this page method just gives me the current page number user is on. How do I get the total number of pages present in the epub file?

**Comment by lzzy12** [comment]
Also the `perccent` seems to be giving wrong information. It seems inconsistent with what is currently on the UI

**Comment by jmgeffroy** [comment]
Hi @lzzy12, That's a good point; thank you for your feedback. Based on your observations, I have to look into this more. Obviously, page counts in reflowable epubs are approximations, but not as bad as what you are experiencing. Today I'm busy, but I'll look at it asap.

**Comment by lzzy12** [comment]
Hi, @jmgeffroy did you get a chance to look at this issue?
